session_2/cricviz/app/models/cricketer.rb

```diff
@@ -68,14 +83,57 @@ def self.import_classical_batters
   # `ActiveRecord::RecordNotFound` exception with the player's name as
   # the message.
   def self.update_innings(batting_scorecard, bowling_scorecard)
-    raise NotImplementedError
+    batting_scorecard.each do |x|
+      if Cricketer.exists?(name: x[0])
```

Comment:
This function makes two SQL queries for each player - one for `exists?` and one for `find_by`.  The problem with SQL queries is that they take a lot of time, often slowing the website.

Can you write this to make just one queries for each player?

There's also a way to write this to make one query _in total_, so try if you can improve it further